src/tests/qualifications/qualification.test.js

```diff
@@ -0,0 +1,55 @@
+const expect = require('expect')
+const { request } = require('../../utils/test')
+const { QualificationFactory } = require('../../utils/factories/')
+const Qualification = require('../../models/qualifications')
+
+describe('qualifications', () => {
+  let skills = []
+  let knowledge = []
+  before(async () => {
+    for (let y = 0; y < 10; y++) {
+      const skill = await Qualification.create(
+        QualificationFactory.generateSkill()
+      )
+      skills.push(`"${skill._id}"`)
+      const knowledge1 = await Qualification.create(
+        QualificationFactory.generateKnowledge()
+      )
+      knowledge.push(`"${knowledge1._id}"`)
+    }
+  })
+
+  it('there is more than 5 skills in database', () => {
```

Comment:
Same goes here

Reply:
it is checking there are enough skills and knowledge for signup to be successfull

Reply:
It's not. You're checking if there are skills added - but not checking if it *can be* added via a mutation

Reply:
There should be a mutation that allows addition of skills. You added the skills using Mongoose, and not with an actual mutation that can be accessed publicly

Reply:
The test should be:
1. Create 3 skills
2. Test creating a user
3. This should fail since there are only 3 skills added
4. Now create 2 additional skills
5. Try and create a user again
6. This should succeed since there are already 5 skills

Reply:
Also, the logic doesn't to make sense to me. You cannot add a user if there are less than 5 skills in the database? Should it be you cannot create a user if it has less than 5 skills added to the user itself? And not just in the database?

Reply:
Based on your code, skills can be added in a user - which i think is correct. But the addition should check for the same:
Users to be created should have at least 5 skills.

Reply:
I assumed each user should have certain skills etc. i.e. required on front end so there should be enough skills in database to be able to user selected that skills in order to signup. User cannot signup with less than particular number of skills. You can check that on apollo playground.

Reply:
You should not depend what the behavior will be in the front-end. This is an API and it can be used in any way the user wants. If our API allows it, whatever application will use it that way.

Also, you mean if there are 5 skills in the database, I can create a user *WITHOUT* any skill? Since the requirement is just to have at least 5 skills in the database?

If yes - you are still not testing it that way since you are just *merely* counting the skills, knowledge in the database and *NOT* checking if a user can be created

Reply:
got it, this is a bug.

Reply:
fixed it